### ISSUE TITLE
fastp: Percentage Reads Passing Filter

### ISSUE THREAD

**Issue by g-pacheco** [ewels/MultiQC#1574]
Hello MultiQC, 

To be very honest, I am not sure if that is a bug from your side.  I have opened [this issue](https://github.com/OpenGene/fastp/issues/367) on fastp, but I now see that this could also be fixed by you {maybe}. The thing is that if I get this right before fastp reported the SAME number for `after_filtering:total_reads` & `filtering_result:passed_filter_reads` as can be seen in their [example](http://opengene.org/fastp/fastp.json). However, I reckon this changed recently, and these numbers are no longer the same as can be seen in my example below: 

```json
                "after_filtering": {
                        "total_reads":1176646,
                        "total_bases":183937612,
                        "q20_bases":179080377,
                        "q30_bases":170016138,
                        "q20_rate":0.973593,
                        "q30_rate":0.924314,
                        "read1_mean_length":156,
                        "gc_content":0.434624
                }
        },
        "filtering_result": {
                "passed_filter_reads": 6312238,
                "corrected_reads": 256148,
                "corrected_bases": 621598,
                "low_quality_reads": 172442,
                "too_many_N_reads": 2,
                "low_complexity_reads": 1930,
                "too_short_reads": 24358,
                "too_long_reads": 0
```

So, I was wondering if MultiQC could be collecting this number from the wrong place. Would it be possible to get it from the `passed_filter_reads` {I believe the correct one}?

Thanks a lot, George.

**Comment by ewels** [comment]
I guess you're talking about the General Statistics table here, the `% PF` (`Percent reads passing filter`) column?

That figure is calculated in this code:

https://github.com/ewels/MultiQC/blob/40a4d182d03260eeadaab8ca0eda2d1e24302ecf/multiqc/modules/fastp/fastp.py#L210-L213

Yeah.. I'm not entirely sure why we don't use the `passed_filter_reads` number. We use that for one of the plots:
https://github.com/ewels/MultiQC/blob/40a4d182d03260eeadaab8ca0eda2d1e24302ecf/multiqc/modules/fastp/fastp.py#L359

So basically I think that you're right about what MultiQC is doing. And I can't see a reason why we couldn't switch to your proposed solution. Fancy opening a pull-request? 😁

**Comment by fgvieira** [comment]
Should we try to get some feedback from @sfchen? Just in case he is planning changes to that output file...

**Comment by g-pacheco** [comment]
Thanks, @ewels! We can proceed with the pull-request, but I agree with *fgvieira* that we should try to get some feedback from *sfchen*.  So let us wait a bit to see if we get any.

**Comment by g-pacheco** [comment]
Hello @ewels,

I have now opened the pull-request [#1587](https://github.com/ewels/MultiQC/pull/1587). It is my first pull-request ever, so I apologise in advance if I have not done it in the most correct way.

Best regards, George.
